physical/raft/raft_autopilot.go

```diff
@@ -0,0 +1,556 @@
+	}
+
+	return ret
+}
+
+func (d *Delegate) RemoveFailedServer(server *autopilot.Server) {
+	// TODO: implement me
+}
+
+// SetFollowerStates sets the followerStates field in the backend to track peers
+// in the raft cluster.
+func (b *RaftBackend) SetFollowerStates(states *FollowerStates) {
+	b.l.Lock()
+	b.followerStates = states
+	b.l.Unlock()
+}
+
+// SetAutopilotConfig updates the autopilot configuration in the backend.
+func (b *RaftBackend) SetAutopilotConfig(config *AutopilotConfig) {
+	b.l.Lock()
+	b.autopilotConfig = config
+	b.l.Unlock()
+}
+
+// AutopilotConfig returns the autopilot configuration in the backend.
+func (b *RaftBackend) AutopilotConfig() *AutopilotConfig {
+	b.l.RLock()
+	conf := b.autopilotConfig
+	b.l.RUnlock()
+	return conf
+}
+
+func (b *RaftBackend) defaultAutopilotConfig() *AutopilotConfig {
+	return &AutopilotConfig{
+		CleanupDeadServers:      false,
+		LastContactThreshold:    10 * time.Second,
+		MaxTrailingLogs:         1000,
+		MinQuorum:               3,
+		ServerStabilizationTime: 10 * time.Second,
+	}
+}
+
+func (b *RaftBackend) autopilotConf() (*AutopilotConfig, error) {
+	config := b.conf["autopilot"]
+	if config == "" {
+		return nil, nil
+	}
+
+	// TODO: Find out why we are getting a list instead of a single item
+	var configs []*AutopilotConfig
+	err := jsonutil.DecodeJSON([]byte(config), &configs)
+	if err != nil {
+		return nil, errwrap.Wrapf("failed to decode autopilot config: {{err}}", err)
+	}
+	if len(configs) != 1 {
+		return nil, fmt.Errorf("expected a single block of autopilot config")
+	}
+
+	return configs[0], nil
+}
+
+// StartAutopilot puts autopilot subsystem to work. This should only be called
+// on the active node.
+func (b *RaftBackend) StartAutopilot(ctx context.Context) {
+	if b.autopilot == nil {
+		return
+	}
+	b.autopilot.Start(ctx)
+}
+
+// StopAutopilot stops a running autopilot instance. This should only be called
+// on the active node.
+func (b *RaftBackend) StopAutopilot() {
+	if b.autopilot == nil {
+		return
+	}
+	b.autopilot.Stop()
+}
+
+// AutopilotState represents the health information retrieved from autopilot.
+type AutopilotState struct {
+	Healthy                    bool `json:"healthy"`
+	FailureTolerance           int  `json:"failure_tolerance"`
+	OptimisticFailureTolerance int  `json:"optimistic_failure_tolerance"`
+
+	Servers   map[string]AutopilotServer `json:"servers"`
+	Leader    string                     `json:"leader"`
+	Voters    []string                   `json:"voters"`
+	NonVoters []string                   `json:"non_voters,omitempty"`
+}
+
+// AutopilotServer represents the health information of individual server node
+// retrieved from autopilot.
+type AutopilotServer struct {
+	ID          string            `json:"id"`
+	Name        string            `json:"name"`
+	Address     string            `json:"address"`
+	NodeStatus  string            `json:"node_status"`
+	LastContact *ReadableDuration `json:"last_contact"`
```

Comment:
This wasn't the issue. Fixed the audit log problem. Using a `*AutopilotServer` instead of `AutopilotServer` in the Servers map under AutopilotState resolved the problem.